### ISSUE TITLE
[Feature]: Equilibrium Matching

### ISSUE THREAD

**Issue by soran-ghaderi** [soran-ghaderi/torchebm#79]
### Prerequisites

- [x] I have searched existing issues and discussions to ensure this feature hasn't been requested before
- [x] I have read the documentation and confirmed this feature doesn't already exist

### Problem Summary

Equilibrium Matching (EqM) learns a time-invariant gradient field that is compatible with an underlying energy function, eliminating time/noise conditioning and fixed-horizon integrators.

https://arxiv.org/pdf/2510.02300

### Proposed Solution

https://arxiv.org/pdf/2510.02300
https://github.com/raywang4/EqM/blob/main/train.py

### Feature Category

Loss Functions

### Priority

High - Important for my use case

### Example Usage

```python
from torchebm.losses import EquilibriumMatching

eq_loss_fn = EquilibriumMatching(
    energy_function=energy_fn,
    device=device,
)


# training:
for epoch in range(10):
    epoch_loss = 0.0
    for batch_data in dataloader:
        batch_data = batch_data.to(device)

        optimizer.zero_grad()
        loss = eq_loss_fn(batch_data)
        loss.backward()
        optimizer.step()
        ...
```

### Implementation

- [ ] I would be willing to contribute to the implementation
- [ ] I can help with testing the feature
- [ ] I can help with documentation
- [ ] I can provide additional use cases/examples

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct
